Dockerfile.postgres.dev

```diff
@@ -1,4 +1,4 @@
-FROM postgres:14
+FROM postgres:15
```

Comment:
maybe do this separately next time, could be easy to miss amidst the bunch of other changes going in

Reply:
Good call - cherry-picked to [another branch](https://github.com/supabase/gotrue/pull/882) that can go in first and should be good after